### ISSUE TITLE
Certain data class attributes not being set as required in Openapi document

### ISSUE THREAD

**Issue by rikurb8** [micronaut-projects/micronaut-openapi#490]
### Steps to Reproduce

1. Create Micronaut project with Micronaut Openapi using Kotlin
2. Add return value for controller function that has non-optional fields
3. Build project to generate Openapi documentation

### Expected Behaviour

Values from data class should be generated in to Openapi documentation correctly, schema required section should contain all attributes not marked nullable

### Actual Behaviour

For certain types, eg. String and custom objects this works fine, but for other types eg. Int and Float the attributes are not added to the `required` section even if declared non-nullable (eg. `val average: Float`). Adding `@SchemaI(required = true)` to the attributes doesn't add them to the required section either. 

### Environment Information

- **Operating System**: MacOS 11.2.3 
- **Micronaut Version:** 2.1.2
- **JDK Version:** openjdk version "1.8.0_265"

### Example Application
https://github.com/rikurb8/micronaut-openapi-kotlin-issue

**Comment by jbreton** [comment]
`Long` and `Boolean` are also affected by this.

**Comment by rikurb8** [comment]
We added `@field:NotBlank` to all required attributes, with that the needed values get marked required automatically.

**Comment by ilopmar** [comment]
Fixed by #540

**Comment by adamkobor** [comment]
> `Long` and `Boolean` are also affected by this.
> 
> The workaround we found is the define those properties in `@Schema(requiredProperties = ["booleanProperty"])`.

We're facing the same issue since version 4.0.0, I assume that #636 might be the reason of it :(
We have a data class like this:
```kotlin
data class LatLongPair(
    val lat: Double,
    val long: Double
)
````

The property called `long` is marked as not required in the generated OpenAPI yml, also a `Boolean` property called `public` in an other data class.
